runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainer.java

```diff
@@ -77,12 +82,72 @@ public FlinkMetricContainer(RuntimeContext runtimeContext) {
     this.metricsAccumulator = (MetricsAccumulator) metricsAccumulator;
   }
 
-  MetricsContainer getMetricsContainer(String stepName) {
+  public MetricsContainer getMetricsContainer(String stepName) {
     return metricsAccumulator != null
         ? metricsAccumulator.getLocalValue().getContainer(stepName)
         : null;
   }
 
+  /**
+   * Parse a {@link MetricName} from a {@link
+   * org.apache.beam.model.fnexecution.v1.BeamFnApi.MonitoringInfoUrns.Enum}
+   *
+   * <p>Should be consistent with {@code parse_namespace_and_name} in monitoring_infos.py
+   *
+   * <p>TODO: not flink-specific; where should it live?
+   */
+  public static MetricName parseUrn(String urn) {
+    if (urn.startsWith(USER_COUNTER_URN_PREFIX.toString())) {
```

Comment:
Does calling toString on this work? The usage I have seen before involved extracting the attached object and pulling out a field from the enum.

See BeamUrns.getUrns
https://github.com/apache/beam/blob/a5a139d5c7be01fbbd1f8b14df9ef5e05696af2c/runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/BeamUrns.java#L27

You could reference this file which pulls it in. SimpleMonitoringInfoBuilder

https://github.com/apache/beam/blob/dc10f75a7bda91f251d43309a21327fc2ab26154/runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SimpleMonitoringInfoBuilder.java#L64

Reply:
good catch, thanks! I just filed #7408 to fix